packages/@aws-cdk/cdk/lib/resource.ts

```diff
@@ -6,9 +6,14 @@ import { PhysicalName } from './physical-name';
  */
 export interface IResource extends IConstruct {
   /**
-   * The physical (that is, visible in the AWS Console) name of this resource.
+   * Returns the physical name, as a String.
+   *
+   * If the user assigned a physical name to the resource, it will be returned,
+   * unchanged; can be `undefined`, if a physical name was not provided, or an
+   * opaque Token, if a generated name should be assigned lazily at synthesis
```

Comment:
This description probably needs updating as well (cannot be undefined, will return a Token, etc.).